tests/components/rfxtrx/test_switch.py

```diff
@@ -147,80 +145,79 @@ async def test_several_switches(hass, rfxtrx):
+    assert state.attributes.get("friendly_name") == "Bath"
 
-async def test_discover_switch(hass, rfxtrx):
-    """Test with discovery of switches."""
+    state = hass.states.get("switch.living")
+    assert state
+    assert state.state == "off"
+    assert state.attributes.get("friendly_name") == "Living"
+
+
+@pytest.mark.parametrize("repetitions", [1, 3])
+async def test_repetitions(hass, rfxtrx, repetitions):
+    """Test signal repetitions."""
     await async_setup_component(
         hass,
         "switch",
-        {"switch": {"platform": "rfxtrx", "automatic_add": True, "devices": {}}},
+        {
+            "switch": {
+                "platform": "rfxtrx",
+                "signal_repetitions": repetitions,
+                "devices": {"0b1100cd0213c7f230010f71": {"name": "Test"}},
+            }
+        },
     )
     await hass.async_block_till_done()
 
-    event = rfxtrx_core.get_rfx_object("0b1100100118cdea02010f70")
-    event.data = bytearray(
-        [0x0B, 0x11, 0x00, 0x10, 0x01, 0x18, 0xCD, 0xEA, 0x01, 0x01, 0x0F, 0x70]
+    await hass.services.async_call(
+        "switch", "turn_on", {"entity_id": "switch.test"}, blocking=True
     )
+    await hass.async_block_till_done()
 
-    await _signal_event(hass, event)
-    entity = rfxtrx_core.RFX_DEVICES["118cdea_2"]
-    assert 1 == len(rfxtrx_core.RFX_DEVICES)
-    assert "<Entity 0b1100100118cdea01010f70: on>" == entity.__str__()
+    assert rfxtrx.transport.send.call_count == repetitions
 
-    await _signal_event(hass, event)
-    assert 1 == len(rfxtrx_core.RFX_DEVICES)
 
-    event = rfxtrx_core.get_rfx_object("0b1100100118cdeb02010f70")
-    event.data = bytearray(
-        [0x0B, 0x11, 0x00, 0x12, 0x01, 0x18, 0xCD, 0xEA, 0x02, 0x00, 0x00, 0x70]
+async def test_discover_switch(hass, rfxtrx):
+    """Test with discovery of switches."""
+    await async_setup_component(
+        hass,
+        "switch",
+        {"switch": {"platform": "rfxtrx", "automatic_add": True, "devices": {}}},
     )
+    await hass.async_block_till_done()
+
+    await _signal_event(hass, "0b1100100118cdea02010f70")
+    state = hass.states.get("switch.0b1100100118cdea02010f70")
+    assert state
+    assert state.state == "on"
 
-    await _signal_event(hass, event)
-    entity = rfxtrx_core.RFX_DEVICES["118cdeb_2"]
-    assert 2 == len(rfxtrx_core.RFX_DEVICES)
-    assert "<Entity 0b1100120118cdea02000070: on>" == entity.__str__()
+    await _signal_event(hass, "0b1100100118cdeb02010f70")
+    state = hass.states.get("switch.0b1100100118cdeb02010f70")
+    assert state
+    assert state.state == "on"
 
     # Trying to add a sensor
-    event = rfxtrx_core.get_rfx_object("0a52085e070100b31b0279")
-    event.data = bytearray(b"\nR\x08^\x07\x01\x00\xb3\x1b\x02y")
-    await _signal_event(hass, event)
-    assert 2 == len(rfxtrx_core.RFX_DEVICES)
+    await _signal_event(hass, "0a52085e070100b31b0279")
+    state = hass.states.get("sensor.0a52085e070100b31b0279")
+    assert state is None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5802827)